src/main/groovy/life/qbic/portal/sampletracking/web/SampleParser.groovy

```diff
@@ -0,0 +1,42 @@
+package life.qbic.portal.sampletracking.web
+
+import life.qbic.datamodel.identifiers.SampleCodeFunctions
+
+/**
+ * <h1>Sample Parser</h1>
+ *
+ * <p>Extracts sample codes from a given input</p>
+ *
+ * @since 1.3.0
+ */
+class SampleParser {
+    private static String COLUMN_SEPARATOR = "\t"
+
+    /**
+     * Extracts QBiC barcodes from a (multi-)line string in TSV format.
+     * The barcode has to be contained in the first column. Header rows are ignored
+     * @param string the QBiC barcodes in TSV format
+     * @return a collection of extracted sample codes
+     */
+    static Collection<String> extractSampleCodes(String string) {
+        Collection<String> sampleCodes = new ArrayList<>()
+        List<String> lines = string.split("\n").toList()
```

Comment:
you could also introduce a LINE_SEPARATOR here for the new line

Reply:
no it has to be the Unix `\n`. This is agreed upon with the labs

Reply:
I do not think that this will change at all. since after all, it is a line separator and there are only 2 available